lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/sample/Sampler.java

```diff
@@ -0,0 +1,36 @@
+package edu.stanford.futuredata.macrobase.analysis.sample;
+
+/**
+ * Abstract class for sampling. Should not modify input.
+ * computeSampleIndices computes and stores the indices of samples.
+ * getSamplingMethod returns the type of the sampler.
+ * getSamples returns an array of samples from the input using the computed indices.
+ */
+public abstract class Sampler {
```

Comment:
Is there a specific reason for this kind of interface? Is it intended to pre-generate sampling indices once and then call `getSample(...)` on the same indices for different data sets?

If not, then you might consider marking something like this:
```
    // Returns an array with sample indices
    abstract int[] computeSampleIndices(int populationSize, double samplingRate);
   
    public double[] getSample(double[] input, double sampleRate) {
        // check inputs here
        int[] sampleIndices = computeSampleIndices(input.size(), sampleRate);
        .....
    }
    
     or with generics

    public <T> T[] getSample(T[] input, double sampleRate) {
        // check inputs here
        int[] sampleIndices = computeSampleIndices(input.size(), sampleRate);
        .....
    }
```